scene/3d/node_3d.cpp

```diff
@@ -314,13 +342,14 @@ void Node3D::set_rotation_edit_mode(RotationEditMode p_mode) {
 	if (data.rotation_edit_mode == p_mode) {
 		return;
 	}
+
 	data.rotation_edit_mode = p_mode;
 
```

Comment:
```suggestion
	if (p_mode != ROTATION_EDIT_MODE_BASIS) {
		data.local_transform.orthogonalize();
		_propagate_transform_changed(this);
		if (data.notify_local_transform) {
			notification(NOTIFICATION_LOCAL_TRANSFORM_CHANGED);
		}
	}
```
So, at a minimum, I think that an implementation like this is needed.